schemas/2015-08-01/Microsoft.Cache.json

```diff
@@ -0,0 +1,160 @@
+                                            "type": "integer"
+                                        },
+                                        {
+                                            "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                                        }
+                                    ],
+                                    "description": "Microsoft.Cache/Redis: along with family, determines capacity of the cache",
+                                    "default": 1
+                                }
+                            },
+                            "required": [
+                                "name",
+                                "family",
+                                "capacity"
+                            ]
+                        },
+                        "shardCount": {
+                            "oneOf": [
+                                {
+                                    "type": "integer"
+                                },
+                                {
+                                    "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                                }
+                            ],
+                            "description": "Microsoft.Cache/Redis: (only if using clustering) number of shards in the redis cluster"
+                        },
+                        "enableNonSslPort": {
+                            "oneOf": [
+                                {
+                                    "type": "boolean"
+                                },
+                                {
+                                    "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                                }
+                            ],
+                            "description": "Microsoft.Cache/Redis if set to true, enables less secure direct access to redis on port 6379 (WITHOUT SSL tunneling)"
+                        },
+                        "redisConfiguration": {
+                            "type": "object"
+                        },
+                        "virtualNetwork": {
+                            "oneOf": [
+                                {
+                                    "$ref": "#/definitions/virtualNetworkId"
+                                },
+                                {
+                                    "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                                }
+                            ],
+                            "description": "Microsoft.Cache/Redis (optional) the ARM resource ID of a Classic Virtual Network for the redis cache to join"
+                        },
+                        "subnet": {
```

Comment:
While that sounds like a good design there were some practical reasons we did the API like this, and the api-version is done now. I think for the sake of this PR we should only consider whether this schema is reasonable given the current API.